get_asteroidcount.py

```diff
@@ -52,22 +52,21 @@ def get_data(self,
             # Now that we have retrieved the data, store it
             self.data = asteroid_count
 
-            access_data = True
+            self.access_data = True
 
         except:
-            access_data = False
+            self.access_data = False
 
     def get_string(self):
 
-        if access_data = False:
-            print("Error: Cannot retrieve from the IAU Minor Planet Center.")
-            print("Check your Internet connection and minorplanetcenter.net. If both are functional, please raise an issue on Github.")
+        if self.access_data = False:
+            return "Error: Cannot retrieve from the IAU Minor Planet Center.\nCheck your Internet connection and minorplanetcenter.net. If both are functional, please raise an issue on Github."
 
         else:
-            print("We have discovered", asteroid_count['total'], "asteroids.")
+            return "We have discovered", asteroid_count['total'], "asteroids."
 
             if large:
```

Comment:
This code is actually unreachable, because the function execution ends after the first `return` statement.

Reply:
Also, `asteroid_count` is not available inside the `get_string` function. This is why we store in the the `self.data` variable, so it becomes accessible from everywhere within the class 🙂

Reply:
Here's my suggestion for how the issues I've pointed out could be fixed:
```python
def get_string(self):

    string = ''

    if self.data_access:
        string = 'Error: Cannot retrieve from the IAU Minor Planet Center.\n'\
            'Check your Internet connection and minorplanetcenter.net. '\
            'If both are functional, please create an issue on Github.'
    else:
        string += f'We have discovered {self.data["total"]} asteroids.'

        if 'large' in self.data.keys():
            string += '\nOf those, there are {self.data["large"]} '\
                'near-Earth asteroids larger than 1 km wide.'
        if 'hazard' in self.data.keys():
            string += '\nCurrently, there are {self.data["hazard"]} '\
                'near-Earth asteroids that are classified as potentially '\
                'hazardous.'
    
    return string
```
Just let me know if anything is unclear or you have questions! 🙂